### PR TITLE
CLI allocate and deploy

### LINKED ISSUE #402: devel and 2.1 docs don't explain how to deploy via CLI.
The docs explain how to commission a node here: 
https://docs.ubuntu.com/maas/2.1/en/manage-cli-common  

but do not show at all how to deploy.  After some further digging I see deploy is mentioned in an example for setting a HWE kernel on this page:
https://docs.ubuntu.com/maas/2.1/en/manage-cli-advanced

But no where does it actually explain how to deploy a node from the CLI, that I can find.  The only "How to Deploy" docs I can find are about using the UI.  When I try to use the deploy command, I get this:
bladernr@critical-maas:~$ maas admin machine deploy -d pbrbtr distro_series=xenial
409 CONFLICT

       Content-Type: text/plain; charset=utf-8
               Date: Mon, 17 Apr 2017 18:28:52 GMT
             Server: TwistedWeb/16.0.0
             Status: 409
  Transfer-Encoding: chunked
               Vary: Cookie
    X-Frame-Options: SAMEORIGIN

Can't start machine: it hasn't been allocated.

So now I know I need to allocate it (deploy does that for me in the web UI, why does deploy not do that via CLI?)

Searching again through the help, I can find nothing that tells me how to use the CLI to allocate the node for deployment.

So, my UI is hosed because of a bug in MAAS 2.2 RC2 when upgrading, and I can't deploy from the CLI because the docs are missing those vital steps.

### PR DESCRIPTION
**Pull request by pmatulis**
Fixes #402

### CODE REVIEW

File: en/intro-concepts.md [hunk 1]
```diff
@@ -304,6 +304,10 @@ Deploy.
 Allocates (reserves) a node to the MAAS user performing the action (and
 currently logged in). Changes a node's status from 'Ready' to 'Allocated'.
 
+With the CLI, it is necessary to perform this action prior to deploying. With
+the web UI it is done automatically for the user. Acquiring in the web UI is
+used for machine reservation.
```
Comment (degville) [inline en/intro-concepts.md]: I know this is a design issue, but I find myself wondering what the difference is between a node being reserved with 'acquire' on the CLI and a machine being reserved from the web UI.

Comment (pmatulis) [inline en/intro-concepts.md]: Why do you think there *is* a difference? You either acquire from the CLI or you acquire from the web UI. What *I* do not like is how the CLI uses the command `allocate` and the web UI uses the action `acquire`. Both lead to a node being left with a status of `Allocated`. It's another example of bad usage of terminology.

Comment (degville) [inline en/intro-concepts.md]: It's only that when we have to write about it, the inconsistancy sticks out to the reader, who may then blame the documentation. But I agree, writing our way around bad terminology isn't the best solution.

Comment (pmatulis) [inline en/intro-concepts.md]: I added more stuff on acquire. We needed it anyway. Wanna take a look? I also am giving preference to 'node' over 'machine'.

**Pull request merged**